lib/endicia.rb

```diff
@@ -373,6 +373,82 @@ def self.carrier_pickup_request(tracking_number, package_location, options = {})
     response
   end
 
+  # Pass in a options hash to get all the rates available.
+  # 
+  # At minimum pass in:
+  #       MailClass: "Domestic" or "International"
+  #       ToPostalCode: ...
+  #       FromPostalCode: ...
+  #       WeightOz: ...
+  #
+  # Also pass in other stuff for more specifics, defaults listed
+  #       DateAdvance
+  #       MailpieceShape
+  #       MailpieceDimensions
+  #         Length
+  #         Width
+  #         Height
+  #       Machinable: True
+  #       LiveAnimalSurcharge: False
+  #       ToCountryCode
+  #       Services
+  #         CertifiedMail: OFF
+  #         COD: OFF
+  #         DeliveryConfirmation: OFF
+  #         ElectronicReturnReceipt: OFF
+  #         InsuredMail: OFF
+  #         RestrictedDelivery: OFF
+  #         ReturnReceipt: OFF
+  #         SignatureConfirmation: OFF
+  #         AdultSignature: OFF
+  #         AdultSignatureRestrictedDelivery: OFF
+  #       CODAmount
+  #       InsuredValue
+  #
+  # Returns a hash in the form:
+  #
+  #     {
+  #       :success => true or false
+  #       :error_message => "the message" or nil
+  #       :response_body => Raw response data from endicia
+  #       :prices => Array of hashes containing price data
+  #     }
+  def self.calculate_postage_rates(options = {})
+    url = "#{label_service_url(options)}/CalculatePostageRatesXML"
+
+    mailpiece_dimensions = options.delete(:MailpieceDimensions)
+
+    services = options.delete(:Services)
+
+    xml = Builder::XmlMarkup.new.PostageRatesRequest do |xl|
```

Comment:
yeah, figured that's what was going on.  could we find another name?  this seems like an ec2 instance size or something